<a href="https://colab.research.google.com/github/tost1n/Tp-laboratorio/blob/main/Tp.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [29]:
import sympy as sp
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns

In [30]:
url="https://raw.githubusercontent.com/tost1n/Tp-laboratorio/refs/heads/main/Datasets%20de%20limpieza/Empleo-AMBA.csv"

DF=pd.read_csv(url)


/tmp/ipykernel_6957/3637845576.py:3: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.
  DF=pd.read_csv(url)


In [31]:
DF.head()


,LINK,provincia_id,departamento_id,clae2,letra,Cantidad_trabajadores,Cantidad_trabajadores_Hombres,Cantidad_trabajadores_Mujeres,Remuneracion_media,Remuneracion_media_Hombres,Remuneracion_media_Mujeres,Remuneracion_mediana,Remuneracion_mediana_Hombres,Remuneracion_mediana_Mujeres,Poblacion_radio,Viviendas_radio
0,1_10_12,2,2007,1.0,A,2,2.0,NaN,150000.0000,150000.000,NaN,150000.00,150000.000,NaN,584,606
1,1_10_12,2,2007,6.0,B,1,1.0,NaN,337377.8100,337377.810,NaN,337377.81,337377.810,NaN,584,606
2,1_10_12,2,2007,10.0,C,4,2.0,2.0,53289.2875,43690.815,62887.76,48271.49,43690.815,62887.76,584,606
3,1_10_12,2,2007,13.0,C,1,NaN,1.0,72176.2000,NaN,72176.20,72176.20,NaN,72176.20,584,606
4,1_10_12,2,2007,21.0,C,1,1.0,NaN,373915.9500,373915.950,NaN,373915.95,373915.950,NaN,584,606


In [32]:
DF.tail()

,LINK,provincia_id,departamento_id,clae2,letra,Cantidad_trabajadores,Cantidad_trabajadores_Hombres,Cantidad_trabajadores_Mujeres,Remuneracion_media,Remuneracion_media_Hombres,Remuneracion_media_Mujeres,Remuneracion_mediana,Remuneracion_mediana_Hombres,Remuneracion_mediana_Mujeres,Poblacion_radio,Viviendas_radio
508984,68613310,6,6861,84.0,O,16,6.0,10.0,209554.436875,454433.251667,62627.148000,69575.290,186903.265,51658.81,1022,392
508985,68613310,6,6861,85.0,P,20,5.0,15.0,140988.221500,142675.562000,140425.774667,117515.380,127509.610,107521.15,1022,392
508986,68613310,6,6861,86.0,Q,7,2.0,5.0,165429.368571,103643.960000,190143.532000,104359.000,103643.960,104359.00,1022,392
508987,68613310,6,6861,91.0,R,1,NaN,1.0,137147.570000,NaN,137147.570000,137147.570,NaN,137147.57,1022,392
508988,68613310,6,6861,94.0,S,4,3.0,1.0,114986.142500,97687.203333,166882.960000,127855.085,125019.960,166882.96,1022,392


In [33]:
DF.dtypes

,0
LINK,object
provincia_id,int64
departamento_id,int64
clae2,float64
letra,object
Cantidad_trabajadores,int64
Cantidad_trabajadores_Hombres,float64
Cantidad_trabajadores_Mujeres,float64
Remuneracion_media,float64
Remuneracion_media_Hombres,float64


In [34]:
DF.describe()

,provincia_id,departamento_id,clae2,Cantidad_trabajadores,Cantidad_trabajadores_Hombres,Cantidad_trabajadores_Mujeres,Remuneracion_media,Remuneracion_media_Hombres,Remuneracion_media_Mujeres,Remuneracion_mediana,Remuneracion_mediana_Hombres,Remuneracion_mediana_Mujeres,Poblacion_radio,Viviendas_radio
count,508989.000000,508989.000000,508921.000000,508989.000000,421001.000000,273953.000000,5.089890e+05,4.210010e+05,2.739530e+05,5.089890e+05,4.210010e+05,2.739530e+05,508989.000000,508989.000000
mean,4.769482,5114.854800,52.100965,4.437935,3.032862,3.577201,1.334810e+05,1.431750e+05,1.176492e+05,1.254887e+05,1.363061e+05,1.129792e+05,1029.983839,357.028519
std,1.846051,2045.946851,26.030957,7.793089,3.987388,5.915825,3.571972e+05,4.749185e+05,1.704005e+05,3.445662e+05,4.662013e+05,1.659409e+05,419.087095,123.433025
min,2.000000,2007.000000,1.000000,1.000000,1.000000,1.000000,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,53.000000,12.000000
25%,2.000000,2091.000000,28.000000,1.000000,1.000000,1.000000,6.891100e+04,7.040095e+04,6.245392e+04,6.669544e+04,6.874188e+04,6.037511e+04,754.000000,278.000000
50%,6.000000,6410.000000,52.000000,2.000000,2.000000,1.000000,9.929042e+04,1.026374e+05,9.143507e+04,9.379298e+04,9.843326e+04,8.786776e+04,946.000000,341.000000
75%,6.000000,6539.000000,74.000000,4.000000,3.000000,3.000000,1.493358e+05,1.578475e+05,1.356229e+05,1.406099e+05,1.501897e+05,1.297743e+05,1225.000000,415.000000
max,6.000000,6861.000000,99.000000,254.000000,186.000000,134.000000,1.029963e+08,1.405377e+08,2.776678e+07,1.029963e+08,1.405377e+08,2.776678e+07,5895.000000,1405.000000


In [35]:
#Consultamos si tiene datos faltantes
DF.isnull().sum()

,0
LINK,0
provincia_id,0
departamento_id,0
clae2,68
letra,68
Cantidad_trabajadores,0
Cantidad_trabajadores_Hombres,87988
Cantidad_trabajadores_Mujeres,235036
Remuneracion_media,0
Remuneracion_media_Hombres,87988


In [36]:
#Borramos los datos nulos que interpretamos como faltantes
print(f"Forma del dataframe antes de borrar las filas con nulos: {DF.shape}")

# Borrar filas si "clae2" o "letra" son nulos
DF_limpio = DF.dropna(subset=["clae2", "letra"])

print(f"Forma del dataframe despues de borrar las filas con nulos: {DF_limpio.shape}")
DF_limpio

Forma del dataframe antes de borrar las filas con nulos: (508989, 16)
Forma del dataframe despues de borrar las filas con nulos: (508921, 16)


,LINK,provincia_id,departamento_id,clae2,letra,Cantidad_trabajadores,Cantidad_trabajadores_Hombres,Cantidad_trabajadores_Mujeres,Remuneracion_media,Remuneracion_media_Hombres,Remuneracion_media_Mujeres,Remuneracion_mediana,Remuneracion_mediana_Hombres,Remuneracion_mediana_Mujeres,Poblacion_radio,Viviendas_radio
0,1_10_12,2,2007,1.0,A,2,2.0,NaN,150000.000000,150000.000000,NaN,150000.000,150000.000,NaN,584,606
1,1_10_12,2,2007,6.0,B,1,1.0,NaN,337377.810000,337377.810000,NaN,337377.810,337377.810,NaN,584,606
2,1_10_12,2,2007,10.0,C,4,2.0,2.0,53289.287500,43690.815000,62887.760000,48271.490,43690.815,62887.76,584,606
3,1_10_12,2,2007,13.0,C,1,NaN,1.0,72176.200000,NaN,72176.200000,72176.200,NaN,72176.20,584,606
4,1_10_12,2,2007,21.0,C,1,1.0,NaN,373915.950000,373915.950000,NaN,373915.950,373915.950,NaN,584,606
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
508984,68613310,6,6861,84.0,O,16,6.0,10.0,209554.436875,454433.251667,62627.148000,69575.290,186903.265,51658.81,1022,392
508985,68613310,6,6861,85.0,P,20,5.0,15.0,140988.221500,142675.562000,140425.774667,117515.380,127509.610,107521.15,1022,392
508986,68613310,6,6861,86.0,Q,7,2.0,5.0,165429.368571,103643.960000,190143.532000,104359.000,103643.960,104359.00,1022,392
508987,68613310,6,6861,91.0,R,1,NaN,1.0,137147.570000,NaN,137147.570000,137147.570,NaN,137147.57,1022,392


In [37]:
#Interpretamos los datos nulos en ciertas columnas como valor 0 y lo cambiamos para poder analizarlos como un numero
columnas_con_nulos = ["Cantidad_trabajadores_Hombres", "Cantidad_trabajadores_Mujeres", "Remuneracion_media_Hombres", "Remuneracion_media_Mujeres", "Remuneracion_mediana_Hombres", "Remuneracion_mediana_Mujeres"]
DF_limpio[columnas_con_nulos] = DF_limpio[columnas_con_nulos].fillna(0)
DF_limpio

/tmp/ipykernel_6957/3520448293.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DF_limpio[columnas_con_nulos] = DF_limpio[columnas_con_nulos].fillna(0)


,LINK,provincia_id,departamento_id,clae2,letra,Cantidad_trabajadores,Cantidad_trabajadores_Hombres,Cantidad_trabajadores_Mujeres,Remuneracion_media,Remuneracion_media_Hombres,Remuneracion_media_Mujeres,Remuneracion_mediana,Remuneracion_mediana_Hombres,Remuneracion_mediana_Mujeres,Poblacion_radio,Viviendas_radio
0,1_10_12,2,2007,1.0,A,2,2.0,0.0,150000.000000,150000.000000,0.000000,150000.000,150000.000,0.00,584,606
1,1_10_12,2,2007,6.0,B,1,1.0,0.0,337377.810000,337377.810000,0.000000,337377.810,337377.810,0.00,584,606
2,1_10_12,2,2007,10.0,C,4,2.0,2.0,53289.287500,43690.815000,62887.760000,48271.490,43690.815,62887.76,584,606
3,1_10_12,2,2007,13.0,C,1,0.0,1.0,72176.200000,0.000000,72176.200000,72176.200,0.000,72176.20,584,606
4,1_10_12,2,2007,21.0,C,1,1.0,0.0,373915.950000,373915.950000,0.000000,373915.950,373915.950,0.00,584,606
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
508984,68613310,6,6861,84.0,O,16,6.0,10.0,209554.436875,454433.251667,62627.148000,69575.290,186903.265,51658.81,1022,392
508985,68613310,6,6861,85.0,P,20,5.0,15.0,140988.221500,142675.562000,140425.774667,117515.380,127509.610,107521.15,1022,392
508986,68613310,6,6861,86.0,Q,7,2.0,5.0,165429.368571,103643.960000,190143.532000,104359.000,103643.960,104359.00,1022,392
508987,68613310,6,6861,91.0,R,1,0.0,1.0,137147.570000,0.000000,137147.570000,137147.570,0.000,137147.57,1022,392


In [38]:
# Verificamos los cambios
DF_limpio.isnull().sum()

,0
LINK,0
provincia_id,0
departamento_id,0
clae2,0
letra,0
Cantidad_trabajadores,0
Cantidad_trabajadores_Hombres,0
Cantidad_trabajadores_Mujeres,0
Remuneracion_media,0
Remuneracion_media_Hombres,0


In [39]:
# Contamos el número de filas duplicadas
duplicadas = DF_limpio.duplicated().sum()
print(f"Número de filas duplicadas: {duplicadas}")

Número de filas duplicadas: 0


In [40]:
# 1. Creamos la columna derivada utilizando DF_limpio
DF_limpio['Brecha_Salarial_Genero'] = (DF_limpio['Remuneracion_mediana_Hombres'] - DF_limpio['Remuneracion_mediana_Mujeres']) / DF_limpio['Remuneracion_mediana_Hombres']

# Reemplazar valores infinitos resultantes de la división por cero con NaN
DF_limpio['Brecha_Salarial_Genero'] = DF_limpio['Brecha_Salarial_Genero'].replace([np.inf, -np.inf], np.nan)

# 2. Verificación de los resultados en pantalla
print("¡Columna de Brecha Salarial creada con éxito!")
display(DF_limpio[['Remuneracion_mediana_Hombres', 'Remuneracion_mediana_Mujeres', 'Brecha_Salarial_Genero']].head())

¡Columna de Brecha Salarial creada con éxito!


/tmp/ipykernel_6957/424674540.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DF_limpio['Brecha_Salarial_Genero'] = (DF_limpio['Remuneracion_mediana_Hombres'] - DF_limpio['Remuneracion_mediana_Mujeres']) / DF_limpio['Remuneracion_mediana_Hombres']
/tmp/ipykernel_6957/424674540.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DF_limpio['Brecha_Salarial_Genero'] = DF_limpio['Brecha_Salarial_Genero'].replace([np.inf, -np.inf], np.nan)


,Remuneracion_mediana_Hombres,Remuneracion_mediana_Mujeres,Brecha_Salarial_Genero
0,150000.000,0.00,1.000000
1,337377.810,0.00,1.000000
2,43690.815,62887.76,-0.439382
3,0.000,72176.20,NaN
4,373915.950,0.00,1.000000


Esta columna es un indicador de desigualdad económica relativa. En lugar de mirar los salarios de hombres y mujeres por separado, esta variable unifica ambos datos en un solo número que mide qué porcentaje menos (o más) ganan las mujeres en comparación con los hombres en un mismo radio censal o departamento del AMBA.

En economía laboral y urbana, es fundamental usar este tipo de indicadores relativos porque permiten comparar zonas muy distintas entre sí. No es lo mismo una diferencia absoluta de $50.000 en un barrio de ingresos bajos que en un barrio de ingresos muy altos; al transformarlo en un porcentaje (un ratio relativo), la comparación se vuelve justa y metodológicamente sólida.

Además, el dataset utiliza la remuneración mediana (en lugar del promedio), lo cual es un gran acierto metodológico del grupo, ya que la mediana no se ve afectada por salarios extremadamente altos o bajos, reflejando mejor la realidad del "trabajador típico" de cada zona.
$$\text{Brecha Salarial} = \frac{\text{Salario Hombres} - \text{Salario Mujeres}}{\text{Salario Hombres}}$$

Si el resultado es 0.20: Significa que la brecha es del 20% en favor de los hombres. Dicho de otra forma, por cada $100 que gana un hombre en ese radio del AMBA, una mujer gana $80.

Si el resultado es 0.00: Significa paridad absoluta (0% de brecha). Ambos géneros tienen exactamente la misma remuneración mediana en esa zona.

Si el resultado es -0.05: Significa una brecha del 5% en favor de las mujeres. En ese punto específico, la remuneración mediana femenina supera a la masculina.

In [41]:
# 1. Agrupamos la variable continua en 3 categorías simétricas usando DF_limpio
DF_limpio['Categoria_Densidad'] = pd.qcut(DF_limpio['Poblacion_radio'], q=3, labels=['Baja Densidad', 'Media Densidad', 'Alta Densidad'])

# 2. Verificación de cuántos radios censales cayeron en cada grupo
print("¡Segmentación por densidad completada!")
print(DF_limpio['Categoria_Densidad'].value_counts())

¡Segmentación por densidad completada!
Categoria_Densidad
Media Densidad    169741
Baja Densidad     169669
Alta Densidad     169511
Name: count, dtype: int64


/tmp/ipykernel_6957/2305072428.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DF_limpio['Categoria_Densidad'] = pd.qcut(DF_limpio['Poblacion_radio'], q=3, labels=['Baja Densidad', 'Media Densidad', 'Alta Densidad'])


Esta variable transforma un dato numérico continuo y "pesado" (la cantidad exacta de habitantes de un radio censal, Poblacion_radio) en una variable categórica u ordinal (Baja, Media y Alta Densidad).

Se utilizó un modelo de lenguaje para adaptar las fórmulas de preprocesamiento de datos a la estructura del DataFrame DF_limpio. Se crearon dos transformaciones: (1) La variable derivada 'Brecha_Salarial_Genero' para analizar las desigualdades relativas, y (2) La categorización 'Categoria_Densidad' mediante cuantiles para segmentar el comportamiento de los patrones de empleo.

In [42]:
clae_mapping = {
    'A': 'Agricultura, ganadería, caza, silvicultura y pesca',
    'B': 'Explotación de minas y canteras',
    'C': 'Industrias manufactureras',
    'D': 'Suministro de electricidad, gas, vapor y aire acondicionado',
    'E': 'Suministro de agua; cloacas, gestión de residuos y descontaminación',
    'F': 'Construcción',
    'G': 'Comercio al por mayor y al por menor; reparación de vehículos automotores y motocicletas',
    'H': 'Transporte y almacenamiento',
    'I': 'Alojamiento y servicios de comida',
    'J': 'Información y comunicación',
    'K': 'Actividades financieras y de seguros',
    'L': 'Actividades inmobiliarias',
    'M': 'Actividades profesionales, científicas y técnicas',
    'N': 'Actividades administrativas y servicios de apoyo',
    'O': 'Administración pública, defensa y seguridad social obligatoria',
    'P': 'Enseñanza',
    'Q': 'Actividades de atención de la salud humana y servicios sociales',
    'R': 'Artes, entretenimiento y recreación',
    'S': 'Otras actividades de servicios',
    'T': 'Servicios de hogares privados como empleadores',
    'U': 'Servicios de organizaciones y organismos extraterritoriales'
}

DF_limpio['clae2_description'] = DF_limpio['letra'].map(clae_mapping)
display(DF_limpio['clae2_description'].value_counts().head(10))

/tmp/ipykernel_6957/391166073.py:25: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DF_limpio['clae2_description'] = DF_limpio['letra'].map(clae_mapping)


,count
clae2_description,
Industrias manufactureras,139031
Actividades administrativas y servicios de apoyo,44335
Comercio al por mayor y al por menor; reparación de vehículos automotores y motocicletas,36739
Transporte y almacenamiento,36724
Información y comunicación,33919
"Actividades profesionales, científicas y técnicas",33443
Construcción,27486
Actividades financieras y de seguros,24494
Otras actividades de servicios,20037


In [43]:
DF_limpio.head()


,LINK,provincia_id,departamento_id,clae2,letra,Cantidad_trabajadores,Cantidad_trabajadores_Hombres,Cantidad_trabajadores_Mujeres,Remuneracion_media,Remuneracion_media_Hombres,Remuneracion_media_Mujeres,Remuneracion_mediana,Remuneracion_mediana_Hombres,Remuneracion_mediana_Mujeres,Poblacion_radio,Viviendas_radio,Brecha_Salarial_Genero,Categoria_Densidad,clae2_description
0,1_10_12,2,2007,1.0,A,2,2.0,0.0,150000.0000,150000.000,0.00,150000.00,150000.000,0.00,584,606,1.000000,Baja Densidad,"Agricultura, ganadería, caza, silvicultura y p..."
1,1_10_12,2,2007,6.0,B,1,1.0,0.0,337377.8100,337377.810,0.00,337377.81,337377.810,0.00,584,606,1.000000,Baja Densidad,Explotación de minas y canteras
2,1_10_12,2,2007,10.0,C,4,2.0,2.0,53289.2875,43690.815,62887.76,48271.49,43690.815,62887.76,584,606,-0.439382,Baja Densidad,Industrias manufactureras
3,1_10_12,2,2007,13.0,C,1,0.0,1.0,72176.2000,0.000,72176.20,72176.20,0.000,72176.20,584,606,NaN,Baja Densidad,Industrias manufactureras
4,1_10_12,2,2007,21.0,C,1,1.0,0.0,373915.9500,373915.950,0.00,373915.95,373915.950,0.00,584,606,1.000000,Baja Densidad,Industrias manufactureras


In [44]:
clae_mapping = {
    'A': 'Agricultura, ganadería, caza, silvicultura y pesca',
    'B': 'Explotación de minas y canteras',
    'C': 'Industrias manufactureras',
    'D': 'Suministro de electricidad, gas, vapor y aire acondicionado',
    'E': 'Suministro de agua; cloacas, gestión de residuos y descontaminación',
    'F': 'Construcción',
    'G': 'Comercio al por mayor y al por menor; reparación de vehículos automotores y motocicletas',
    'H': 'Transporte y almacenamiento',
    'I': 'Alojamiento y servicios de comida',
    'J': 'Información y comunicación',
    'K': 'Actividades financieras y de seguros',
    'L': 'Actividades inmobiliarias',
    'M': 'Actividades profesionales, científicas y técnicas',
    'N': 'Actividades administrativas y servicios de apoyo',
    'O': 'Administración pública, defensa y seguridad social obligatoria',
    'P': 'Enseñanza',
    'Q': 'Actividades de atención de la salud humana y servicios sociales',
    'R': 'Artes, entretenimiento y recreación',
    'S': 'Otras actividades de servicios',
    'T': 'Servicios de hogares privados como empleadores',
    'U': 'Servicios de organizaciones y organismos extraterritoriales'
}

DF_limpio['clae2_description'] = DF_limpio['letra'].map(clae_mapping)
display(DF_limpio['clae2_description'].value_counts().head(10))

/tmp/ipykernel_6957/391166073.py:25: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DF_limpio['clae2_description'] = DF_limpio['letra'].map(clae_mapping)


,count
clae2_description,
Industrias manufactureras,139031
Actividades administrativas y servicios de apoyo,44335
Comercio al por mayor y al por menor; reparación de vehículos automotores y motocicletas,36739
Transporte y almacenamiento,36724
Información y comunicación,33919
"Actividades profesionales, científicas y técnicas",33443
Construcción,27486
Actividades financieras y de seguros,24494
Otras actividades de servicios,20037


In [45]:
df_jurisdicciones = pd.read_csv('https://raw.githubusercontent.com/tost1n/Tp-laboratorio/refs/heads/main/jurisdicciones.csv')
display(df_jurisdicciones.head())

,provincia_id,provincia,departamento_id,departamento
0,2,CABA,2007,Comuna 1
1,2,CABA,2014,Comuna 2
2,2,CABA,2021,Comuna 3
3,2,CABA,2028,Comuna 4
4,2,CABA,2035,Comuna 5


In [46]:
DF_limpio = pd.merge(DF_limpio, df_jurisdicciones[['provincia_id', 'provincia', 'departamento_id', 'departamento']],
                     on=['provincia_id', 'departamento_id'],
                     how='left', suffixes=('_DF_limpio', '_df_jurisdicciones'))
DF_limpio.head()

,LINK,provincia_id,departamento_id,clae2,letra,Cantidad_trabajadores,Cantidad_trabajadores_Hombres,Cantidad_trabajadores_Mujeres,Remuneracion_media,Remuneracion_media_Hombres,...,Remuneracion_mediana,Remuneracion_mediana_Hombres,Remuneracion_mediana_Mujeres,Poblacion_radio,Viviendas_radio,Brecha_Salarial_Genero,Categoria_Densidad,clae2_description,provincia,departamento
0,1_10_12,2,2007,1.0,A,2,2.0,0.0,150000.0000,150000.000,...,150000.00,150000.000,0.00,584,606,1.000000,Baja Densidad,"Agricultura, ganadería, caza, silvicultura y p...",CABA,Comuna 1
1,1_10_12,2,2007,6.0,B,1,1.0,0.0,337377.8100,337377.810,...,337377.81,337377.810,0.00,584,606,1.000000,Baja Densidad,Explotación de minas y canteras,CABA,Comuna 1
2,1_10_12,2,2007,10.0,C,4,2.0,2.0,53289.2875,43690.815,...,48271.49,43690.815,62887.76,584,606,-0.439382,Baja Densidad,Industrias manufactureras,CABA,Comuna 1
3,1_10_12,2,2007,13.0,C,1,0.0,1.0,72176.2000,0.000,...,72176.20,0.000,72176.20,584,606,NaN,Baja Densidad,Industrias manufactureras,CABA,Comuna 1
4,1_10_12,2,2007,21.0,C,1,1.0,0.0,373915.9500,373915.950,...,373915.95,373915.950,0.00,584,606,1.000000,Baja Densidad,Industrias manufactureras,CABA,Comuna 1


In [47]:
DF_Clave= DF_limpio[[
    "departamento_id",
    "departamento",
    "Cantidad_trabajadores",
    "Cantidad_trabajadores_Hombres",
    "Cantidad_trabajadores_Mujeres",
    "Remuneracion_media",
    "Remuneracion_media_Hombres",
    "Remuneracion_media_Mujeres",
    "Remuneracion_mediana",
    "Remuneracion_mediana_Hombres",
    "Remuneracion_mediana_Mujeres",
    "Brecha_Salarial_Genero",
     "clae2",
    "clae2_description",

]]

Definimos las variables que nos van a ser utiles en el analisis

In [48]:
DF_Clave.head().round(2)

,departamento_id,departamento,Cantidad_trabajadores,Cantidad_trabajadores_Hombres,Cantidad_trabajadores_Mujeres,Remuneracion_media,Remuneracion_media_Hombres,Remuneracion_media_Mujeres,Remuneracion_mediana,Remuneracion_mediana_Hombres,Remuneracion_mediana_Mujeres,Brecha_Salarial_Genero,clae2,clae2_description
0,2007,Comuna 1,2,2.0,0.0,150000.00,150000.00,0.00,150000.00,150000.00,0.00,1.00,1.0,"Agricultura, ganadería, caza, silvicultura y p..."
1,2007,Comuna 1,1,1.0,0.0,337377.81,337377.81,0.00,337377.81,337377.81,0.00,1.00,6.0,Explotación de minas y canteras
2,2007,Comuna 1,4,2.0,2.0,53289.29,43690.82,62887.76,48271.49,43690.82,62887.76,-0.44,10.0,Industrias manufactureras
3,2007,Comuna 1,1,0.0,1.0,72176.20,0.00,72176.20,72176.20,0.00,72176.20,NaN,13.0,Industrias manufactureras
4,2007,Comuna 1,1,1.0,0.0,373915.95,373915.95,0.00,373915.95,373915.95,0.00,1.00,21.0,Industrias manufactureras


In [49]:
DF_Con_brecha=DF_Con_brecha = DF_Clave[
    (DF_Clave['Cantidad_trabajadores_Hombres'] > 0) &
    (DF_Clave['Cantidad_trabajadores_Mujeres'] > 0)
]


se aplico el filtro ***>0*** en ambas columnas para averiguar con que columnas contabamos para realizar la brecha salarial. Porque hubo casos en los cuales no habia hombres o mujeres, y arrojaba como resultado ***-inf*** o ***1*** en la brecha salarial.

In [50]:
DF_Con_brecha.info()

<class 'pandas.core.frame.DataFrame'>
Index: 186193 entries, 2 to 508920
Data columns (total 14 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   departamento_id                186193 non-null  int64  
 1   departamento                   186193 non-null  object 
 2   Cantidad_trabajadores          186193 non-null  int64  
 3   Cantidad_trabajadores_Hombres  186193 non-null  float64
 4   Cantidad_trabajadores_Mujeres  186193 non-null  float64
 5   Remuneracion_media             186193 non-null  float64
 6   Remuneracion_media_Hombres     186193 non-null  float64
 7   Remuneracion_media_Mujeres     186193 non-null  float64
 8   Remuneracion_mediana           186193 non-null  float64
 9   Remuneracion_mediana_Hombres   186193 non-null  float64
 10  Remuneracion_mediana_Mujeres   186193 non-null  float64
 11  Brecha_Salarial_Genero         185713 non-null  float64
 12  clae2                          1861

Las observaciones con las cuales se puede realizar la brecha salarial son 186193 de 508.921 registros, aproximadamene el 36,6% de los registros presentaba trabajadores de ambos géneros

In [51]:
DF_Con_brecha.head(1000000).round(2)

,departamento_id,departamento,Cantidad_trabajadores,Cantidad_trabajadores_Hombres,Cantidad_trabajadores_Mujeres,Remuneracion_media,Remuneracion_media_Hombres,Remuneracion_media_Mujeres,Remuneracion_mediana,Remuneracion_mediana_Hombres,Remuneracion_mediana_Mujeres,Brecha_Salarial_Genero,clae2,clae2_description
2,2007,Comuna 1,4,2.0,2.0,53289.29,43690.82,62887.76,48271.49,43690.82,62887.76,-0.44,10.0,Industrias manufactureras
13,2007,Comuna 1,4,2.0,2.0,190019.08,273264.70,106773.45,165963.88,273264.70,106773.45,0.61,42.0,Construcción
16,2007,Comuna 1,17,11.0,6.0,61061.61,72858.31,39434.33,58596.49,77724.22,37287.56,0.52,47.0,Comercio al por mayor y al por menor; reparaci...
17,2007,Comuna 1,4,3.0,1.0,91028.99,90027.39,94033.78,112981.86,131929.95,94033.78,0.29,49.0,Transporte y almacenamiento
19,2007,Comuna 1,5,4.0,1.0,228995.35,262164.69,96318.00,96318.00,90092.96,96318.00,-0.07,52.0,Transporte y almacenamiento
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
508915,6861,Vicente López,3,1.0,2.0,48818.08,50974.76,47739.74,49349.95,50974.76,47739.74,0.06,82.0,Actividades administrativas y servicios de apoyo
508916,6861,Vicente López,16,6.0,10.0,209554.44,454433.25,62627.15,69575.29,186903.26,51658.81,0.72,84.0,"Administración pública, defensa y seguridad so..."
508917,6861,Vicente López,20,5.0,15.0,140988.22,142675.56,140425.77,117515.38,127509.61,107521.15,0.16,85.0,Enseñanza
508918,6861,Vicente López,7,2.0,5.0,165429.37,103643.96,190143.53,104359.00,103643.96,104359.00,-0.01,86.0,Actividades de atención de la salud humana y s...


In [52]:
DF_sin_brecha = DF_Clave[
    (DF_Clave['Cantidad_trabajadores_Hombres'] == 0) |
    (DF_Clave['Cantidad_trabajadores_Mujeres'] == 0)
]

In [53]:
#Tabla para los datos que no se le pueden realizar la brecha salarial
DF_sin_brecha.head()

,departamento_id,departamento,Cantidad_trabajadores,Cantidad_trabajadores_Hombres,Cantidad_trabajadores_Mujeres,Remuneracion_media,Remuneracion_media_Hombres,Remuneracion_media_Mujeres,Remuneracion_mediana,Remuneracion_mediana_Hombres,Remuneracion_mediana_Mujeres,Brecha_Salarial_Genero,clae2,clae2_description
0,2007,Comuna 1,2,2.0,0.0,150000.00,150000.00,0.0,150000.00,150000.00,0.0,1.0,1.0,"Agricultura, ganadería, caza, silvicultura y p..."
1,2007,Comuna 1,1,1.0,0.0,337377.81,337377.81,0.0,337377.81,337377.81,0.0,1.0,6.0,Explotación de minas y canteras
3,2007,Comuna 1,1,0.0,1.0,72176.20,0.00,72176.2,72176.20,0.00,72176.2,NaN,13.0,Industrias manufactureras
4,2007,Comuna 1,1,1.0,0.0,373915.95,373915.95,0.0,373915.95,373915.95,0.0,1.0,21.0,Industrias manufactureras
5,2007,Comuna 1,1,1.0,0.0,71575.93,71575.93,0.0,71575.93,71575.93,0.0,1.0,24.0,Industrias manufactureras


In [54]:
DF_sin_brecha.info()

<class 'pandas.core.frame.DataFrame'>
Index: 322728 entries, 0 to 508919
Data columns (total 14 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   departamento_id                322728 non-null  int64  
 1   departamento                   322728 non-null  object 
 2   Cantidad_trabajadores          322728 non-null  int64  
 3   Cantidad_trabajadores_Hombres  322728 non-null  float64
 4   Cantidad_trabajadores_Mujeres  322728 non-null  float64
 5   Remuneracion_media             322728 non-null  float64
 6   Remuneracion_media_Hombres     322728 non-null  float64
 7   Remuneracion_media_Mujeres     322728 non-null  float64
 8   Remuneracion_mediana           322728 non-null  float64
 9   Remuneracion_mediana_Hombres   322728 non-null  float64
 10  Remuneracion_mediana_Mujeres   322728 non-null  float64
 11  Brecha_Salarial_Genero         233448 non-null  float64
 12  clae2                          3227

In [55]:

#El procedimiento consiste en tomar todos los registros que pertenecen a un mismo departamento y promediar sus remuneraciones. Los resultados se ordenan de mayor a menor para facilitar la comparación.
#Este agrupamiento es clave para responder la pregunta central del trabajo: si los departamentos con mayor diversidad económica tienden a pagar mejores salarios que aquellos especializados en pocas actividades.

# Agrupación 1 — Remuneración media por departamento
groupby_salario = (
    DF_Clave.groupby('departamento_id')['Remuneracion_media']
    .mean()
    .sort_values(ascending=False)
    .reset_index()
)
groupby_salario.columns = ['Departamento', 'Remuneración Media ($)']
groupby_salario['Remuneración Media ($)'] = groupby_salario['Remuneración Media ($)'].round(2)

# Tabla prolija
print("=== TOP 10 — Departamentos con mayor remuneración media ===")
display(groupby_salario.head(10).style
    .format({'Remuneración Media ($)': '${:,.2f}'})
    .background_gradient(subset=['Remuneración Media ($)'], cmap='Greens')
    .set_properties(**{'text-align': 'center'})
)

=== TOP 10 — Departamentos con mayor remuneración media ===


,Departamento,Remuneración Media ($)
0,2014,"$225,397.30"
1,2098,"$224,149.60"
2,2091,"$215,694.90"
3,2007,"$189,571.38"
4,6756,"$178,264.60"
5,6861,"$177,206.50"
6,2084,"$175,607.19"
7,2042,"$174,563.46"
8,2077,"$157,386.49"
9,2105,"$150,345.86"


In [56]:
groupby_rubros = (
    DF_Clave.groupby(['departamento_id', 'departamento'])['clae2_description']
    .nunique()
    .sort_values(ascending=False)
    .reset_index()
)

groupby_rubros.columns = ['Departamento ID', 'Departamento', 'Cantidad de Rubros']

print("=== Departamentos ordenados por cantidad de rubros ===")
display(groupby_rubros.head(10).style
    .background_gradient(subset=['Cantidad de Rubros'], cmap='Blues')
    .set_properties(**{'text-align': 'center'})
)

=== Departamentos ordenados por cantidad de rubros ===


,Departamento ID,Departamento,Cantidad de Rubros
0,2007,Comuna 1,21
1,2021,Comuna 3,21
2,2042,Comuna 6,21
3,6028,Almirante Brown,21
4,2077,Comuna 11,21
5,2070,Comuna 10,21
6,2063,Comuna 9,21
7,2105,Comuna 15,21
8,2098,Comuna 14,21
9,2091,Comuna 13,21


In [57]:
print("=== TOP 10 — Departamentos con menor cantidad de rubros ===")
display(groupby_rubros.tail(10).style
    .background_gradient(subset=['Cantidad de Rubros'], cmap='Blues_r')
    .set_properties(**{'text-align': 'center'})
)

=== TOP 10 — Departamentos con menor cantidad de rubros ===


,Departamento ID,Departamento,Cantidad de Rubros
41,6638,Pilar,20
42,6648,Presidente Perón,20
43,6840,Tres de Febrero,20
44,6805,Tigre,20
45,6756,San Isidro,20
46,6760,San Miguel,20
47,6861,Vicente López,20
48,6098,Berisso,19
49,6119,Brandsen,19
50,6329,General Las Heras,19


In [58]:
groupby_brecha_salarial = (
    DF_Clave.groupby(['departamento_id', 'departamento'])['Brecha_Salarial_Genero']
    .mean()
    .reset_index()
)

groupby_brecha_salarial.columns = ['Departamento ID', 'Departamento', 'Brecha Salarial Promedio']

# Combinar con la tabla groupby_rubros existente
tabla_final = pd.merge(groupby_rubros, groupby_brecha_salarial[['Departamento ID', 'Brecha Salarial Promedio']], on='Departamento ID', how='left')

correlation = tabla_final['Cantidad de Rubros'].corr(tabla_final['Brecha Salarial Promedio'])

print(f"Correlación entre Cantidad de Rubros y Brecha Salarial Promedio: {correlation:.2f}")

Correlación entre Cantidad de Rubros y Brecha Salarial Promedio: -0.48


La correlación entre la cantidad de rubros y la brecha salarial promedio por departamento es de -0.48. Esto indica una correlación negativa moderada, lo que sugiere que a medida que aumenta la cantidad de rubros en un departamento, la brecha salarial de género tiende a disminuir.

In [59]:
tabla_final.head().round(2)

,Departamento ID,Departamento,Cantidad de Rubros,Brecha Salarial Promedio
0,2007,Comuna 1,21,-11908.65
1,2021,Comuna 3,21,-5483.27
2,2042,Comuna 6,21,-6308.68
3,6028,Almirante Brown,21,-2226.25
4,2077,Comuna 11,21,-958.03
